src/commands/force/user/create.ts

```diff
@@ -184,48 +212,29 @@ export class UserCreateCommand extends SfdxCommand {
 
     if (this.varargs) {
       Object.keys(this.varargs).forEach((key) => {
-        defaultFields[this.lowerFirstLetter(key)] = this.varargs[key];
-
         if (key.toLowerCase() === 'generatepassword') {
-          defaultFields['generatePassword'] = this.varargs[key];
+          // standardize generatePassword casing
+          defaultFields['generatePassword'] = standardizePasswordToBoolean(this.varargs[key]);
+        } else if (key.toLowerCase() === 'profilename') {
+          // standardize profileName casing
+          defaultFields['profileName'] = this.varargs[key];
+        } else {
+          // all other varargs are left "as is"
+          defaultFields[this.lowerFirstLetter(key)] = this.varargs[key];
         }
       });
     }
 
     // check if "profileName" was passed, this needs to become a profileId before calling User.create
     if (defaultFields['profileName']) {
-      const name = (defaultFields['profileName'] || 'Standard User') as string;
+      const name = (defaultFields['profileName'] ?? 'Standard User') as string;
       this.logger.debug(`Querying org for profile name [${name}]`);
       const response: QueryResult<{ Id: string }> = await this.org
```

Comment:
@mshanemc should test this with user that does not have access to profile standard object. Chatter user for example.